application.go

```diff
@@ -109,6 +129,16 @@ type DeleteAppOpts struct {
 	Force bool `url:"force,omitempty"`
 }
 
+// SetIPAddressPerTask defines that the application will have a IP address defines by a external agent.
+// This configuration is not allowed to be used with Port or PortDefinitions.
+func (r *Application) SetIPAddressPerTask(ipAddressPerTask *IPAddressPerTask) *Application {
```

Comment:
If you pass non pointer the client will always send this field in the json body, thus marathon will think it's a default IP-per-task configuration. Using it as a pointer will avoid this behavior.